File: Linguagem de Programação I/Extras/Beecrowd_Extra/1047.ts
import * as readline from 'readline';

let rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question('', (input: string) => {
    let times: number[] = input.split(' ').map(Number);
    let [sh, sm, eh, em] = times;

    let st = sh * 60 + sm;
    let et = eh * 60 + em;
    let d = et - st;
    if (d <= 0) {
        d += 24 * 60;
    }
    let h = Math.floor(d / 60);
    let m = d % 60;
    console.log(`O JOGO DUROU ${h} HORA(S) E ${m} MINUTO(S)`);
    rl.close();
});

File: Linguagem de Programação I/Extras/Beecrowd_Extra/1047.rb
times = gets.split.map(&:to_i)
sh, sm, eh, em = times[0], times[1], times[2], times[3]

st = sh * 60 + sm
et = eh * 60 + em
d = et - st
if d <= 0
    d += 24 * 60
end
h = d / 60
m = d % 60
puts "O JOGO DUROU #{h} HORA(S) E #{m} MINUTO(S)"